### ISSUE TITLE
Training with DAMON dataset, and using RICH dataset images

### ISSUE THREAD

**Issue by EnnaSachdeva** [sha2nkt/deco#16]
Hi, Thank you for this valuable work and your contributions.

I get this error when training the model with the given configs/cfg_train.yml file with all 3 datasets: rich, damon and prox. Could you please suggest the resolution?

![abs](https://github.com/sha2nkt/deco/assets/15670305/d33a31f6-ae78-4214-9796-94ff3fbd847b)

If I train with damon only, I get this error.

![damon](https://github.com/sha2nkt/deco/assets/15670305/3df49fc9-33e6-49cf-b44b-aad42c62b6a2)

**Comment by ac5113** [comment]
Hello,

For training using RICH and PROX, you would have to download the data from their corresponding websites ([RICH](https://rich.is.tue.mpg.de/), [PROX](https://prox.is.tue.mpg.de/)). The error likely occurs since the data is missing, and we do not provide the npzs or the data we have used for training.

As for DAMON, we have updated the [download section](https://deco.is.tue.mpg.de/download.php) to provide the links for the 2D polygon contacts from HOT. Apologies for this oversight. With these files downloaded, the PAL loss term should work again.

**Comment by ac5113** [comment]
I shall close this issue if you're able to train the model using the dataset now.

**Comment by EnnaSachdeva** [comment]
@ac5113 , with the following changes in config_train.py which means it uses only damon data for training, it still gives the errors related to RICH dataset.

![image](https://github.com/sha2nkt/deco/assets/15670305/e1b973e5-ffb8-488a-be0a-0621a76eca3a)
`
![image](https://github.com/sha2nkt/deco/assets/15670305/faf18557-1c76-46d2-86ec-8737bae5aeb0)

**Comment by ac5113** [comment]
You should remove 'rich' and 'prox' altogether from the DATASETS list. The way the loader was written, all the mentioned datasets are loaded and then the dataloader is created by picking from the datasets according to the mentioned probabilities.

`DATASETS: ['damon']` would work

**Comment by EnnaSachdeva** [comment]
It works now. Thank you so much!

**Comment by Sharpiless** [comment]
Hello, awesome work!

May I ask you how to prepare RICH and PROX for training?

**Comment by Sharpiless** [comment]
One more question, do the results in the paper trained on seperate or mixed datasets?

**Comment by ac5113** [comment]
> Hello, awesome work!
> 
> May I ask you how to prepare RICH and PROX for training?

You can use the instructions specified [here](https://github.com/sha2nkt/deco/tree/main?tab=readme-ov-file#training-on-custom-datasets) to prepare the RICH and PROX datasets for training.


> One more question, do the results in the paper trained on seperate or mixed datasets?

The results are on a mixture of RICH, PROX and DAMON datasets

**Comment by Sharpiless** [comment]
Thanks!

**Comment by Sharpiless** [comment]
Sorry for bothering you again for "DAMON challenge: 3D contact prediction from 2D images". Should I provide a result with higher Geo_Err as the arrow indicates? In https://codalab.lisn.upsaclay.fr/competitions/17561#results, higher 
Geo_Err leads to a higher ranking. But I think this might be a typo.

**Comment by ac5113** [comment]
~~The arrow seems to only be for sorting purposes. Results with a lower geodesic error are better since the metric measures the deviation between the predicted and gt contact vertices~~

The results are compared according to the criteria explained on the website as elucidated by @sha2nkt below

**Comment by Sharpiless** [comment]
Thanks again. Any evaluation results of DECO trained on DAMON dataset only? Considering the large storage space occupied by the other two datasets, we plan to train with only a single dataset for the competition.

**Comment by ac5113** [comment]
There was no training done on DAMON only, unfortunately. Thus we do not have these results.
However, you are free to use any other additional training data if RICH and PROX are too bulky.

**Comment by sha2nkt** [comment]
@Sharpiless The ranking order on the CodaLab server is not meant to be the final ranking. As indicated on the challenge [website](https://codalab.lisn.upsaclay.fr/competitions/17561#learn_the_details-evaluation), after the competition ends, we will average the rankings on "F1 Score" and "Geo Error" to determine the winners.
